src/adapter/src/catalog/open.rs

```diff
@@ -291,29 +323,48 @@ impl Catalog {
 
         let mut builtin_table_updates = Vec::new();
 
+        // Make life easier by consolidating all updates, so that we end up with only positive
+        // diffs.
+        let commit_ts = txn.commit_ts();
+        let mut updates = into_consolidatable_updates(updates, commit_ts);
+        differential_dataflow::consolidation::consolidate_updates(&mut updates);
+        soft_assert_no_log!(
```

Comment:
My understanding of `soft_assert_no_log` is that it won't log in prod. I opted to use `soft_assert_no_log` for the assertions that iterate through a `Vec` to avoid the runtime cost since it's expected to always be true. Though maybe that's a premature optimization.